src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.OverlappedValueTaskSource.Windows.cs

```diff
@@ -189,15 +201,87 @@ private static void IOCallback(uint errorCode, uint numBytes, NativeOverlapped*
 
             internal void Complete(uint errorCode, uint numBytes)
             {
+                ReadOnlyMemory<byte> buffer = _buffer;
+                IReadOnlyList<ReadOnlyMemory<byte>>? buffers = _writeBuffers;
                 ReleaseResources();
 
                 switch (errorCode)
                 {
                     case 0:
+                        // Success
+                        switch (_operation)
+                        {
+                            case RandomAccess.Operation.Read:
+                            case RandomAccess.Operation.ReadScatter:
+                            case RandomAccess.Operation.Write when numBytes == 0 || numBytes == buffer.Length:
+                            case RandomAccess.Operation.WriteGather when numBytes == 0 || numBytes == RandomAccess.CountBuffersLength(buffers!):
+                                // Successfully completed a read, or successfully wrote everything needed.
+                                _source.SetResult((int)numBytes);
+                                break;
+
+                            default:
+                                // Successfully completed a write but for fewer bytes than requested be written.
```

Comment:
It seems like it is possible, at least for the disk-full case.

However, since FileStream has ignored such cases for forever, I'll back out the changes to handle the case on Windows for now.  We can add it later if necessary.